pkg/porter/explain_test.go

```diff
@@ -38,14 +38,37 @@ func TestExplain_generateTable(t *testing.T) {
 	err = opts.Validate([]string{}, p.Context)
 	require.NoError(t, err)
 
-	err = p.printBundleExplain(opts, pb)
+	err = p.printBundleExplain(opts, pb, b)
 	assert.NoError(t, err)
 	gotOutput := p.TestConfig.TestContext.GetOutput()
 	expected, err := ioutil.ReadFile("testdata/explain/expected-table-output.txt")
 	require.NoError(t, err)
 	assert.Equal(t, string(expected), gotOutput)
 }
 
+func TestExplain_generateTableRequireDocker(t *testing.T) {
+	p := NewTestPorter(t)
+	defer p.Teardown()
+
+	p.TestConfig.TestContext.AddTestFile("testdata/explain/bundle-docker.json", "bundle-docker.json")
+	b, err := p.CNAB.LoadBundle("bundle-docker.json")
+
+	pb, err := generatePrintable(b, "")
+	require.NoError(t, err)
+	opts := ExplainOpts{}
+	opts.RawFormat = "plaintext"
+
+	err = opts.Validate([]string{}, p.Context)
+	require.NoError(t, err)
+
+	err = p.printBundleExplain(opts, pb, b)
+	assert.NoError(t, err)
+	gotOutput := p.TestConfig.TestContext.GetOutput()
+	expected, err := ioutil.ReadFile("testdata/explain/expected-table-output-docker.txt")
```

Comment:
We have a test helper that makes it easier to compare command output with the contents of a file and keep it in sync when the command output changes.

```
import "get.porter.sh/porter/pkg/test"

test.CompareGoldenFile(t, "testdata/explain/expected-table-output-docker.txt", gotOutput)
```

If you run `mage updateTestFiles` it will find any tests whose output has changed and updates the testdata file automatically so the test passes.

We aren't using that new helper function everywhere in Porter yet, but I'd like new tests to use it so they are easier to maintain.